### PR TITLE
[settings] reformat doxygen

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

**Comment by bukepo** [comment]
thanks for the cleanup, could you please also remove those not used functions?
```
otPlatSettingsBeginChange()
otPlatSettingsCommitChange()
otPlatSettingsAbandonChange()
```

**Comment by zhanglongxia** [comment]
@bukepo  Already submitted a PR #3971 to remove these useless functions.

File: include/openthread/platform/settings.h [hunk 1]
```diff
@@ -149,24 +128,15 @@ otError otPlatSettingsSet(otInstance *aInstance, uint16_t aKey, const uint8_t *a
...
- *
- * @retval OT_ERROR_NONE
- *         The given setting was added or staged to be added.
- * @retval OT_ERROR_NOT_IMPLEMENTED
- *         This function is not implemented on this platform.
- * @retval OT_ERROR_NO_BUFS
- *         No space remaining to store the given setting.
+ * @param[in]     aInstance     The OpenThread instance structure.
+ * @param[in]     aKey          The key associated with the setting to change.
+ * @param[out]    aValue        A pointer to where the new value of the setting should be read from. MUST NOT be NULL
```
Comment (bukepo) [inline include/openthread/platform/settings.h]: should be `[in]`

File: include/openthread/platform/settings.h [hunk 2]
```diff
@@ -114,24 +102,15 @@ otError otPlatSettingsGet(otInstance *aInstance, uint16_t aKey, int aIndex, uint
...
- *
- *  @retval OT_ERROR_NONE
- *          The given setting was changed or staged.
- *  @retval OT_ERROR_NOT_IMPLEMENTED
- *          This function is not implemented on this platform.
- *  @retval OT_ERROR_NO_BUFS
- *          No space remaining to store the given setting.
+ *  @param[in]  aInstance     The OpenThread instance structure.
+ *  @param[in]  aKey          The key associated with the setting to change.
+ *  @param[out] aValue        A pointer to where the new value of the setting should be read from. MUST NOT be NULL if
```
Comment (bukepo) [inline include/openthread/platform/settings.h]: should be `[in]`

File: include/openthread/platform/settings.h (hunk 1 above)
Comment (zhanglongxia) [reply include/openthread/platform/settings.h]: Updated

File: include/openthread/platform/settings.h (hunk 2 above)
Comment (zhanglongxia) [reply include/openthread/platform/settings.h]: Updated

**Pull request merged**